src/main/java/com/atherys/towns/facade/PlotBorderFacade.java

```diff
@@ -0,0 +1,183 @@
+            addSelectionBorder(player, borderInfo);
+        }
+    }
+
+    public void initBorderTask() {
+        Task.builder().execute(task -> activeBorders.forEach((uuid, borderInfoSet) -> {
+            Sponge.getServer().getPlayer(uuid).ifPresent(player -> {
+                borderInfoSet.forEach(borderInfo -> {
+                    int xLength = MathUtils.getXLength(borderInfo.getNECorner(), borderInfo.getSWCorner());
+                    int zLength = MathUtils.getZLength(borderInfo.getNECorner(), borderInfo.getSWCorner());
+                    Vector3d particleLocationNE = new Vector3d(borderInfo.getNECorner().getX(), player.getPosition().getFloorY(), borderInfo.getNECorner().getY());
+                    Vector3d particleLocationSW = new Vector3d(borderInfo.getSWCorner().getX(), player.getPosition().getFloorY(), borderInfo.getSWCorner().getY());
+
+                    player.spawnParticles(borderInfo.getEffect(), particleLocationNE.add(1, 0, 0));
+                    player.spawnParticles(borderInfo.getEffect(), particleLocationSW.add(0, 0, 1));
+                    for (int i = 0; i <= zLength; i++) {
+                        player.spawnParticles(borderInfo.getEffect(), particleLocationNE.add(1, 0, i + 1));
+                        player.spawnParticles(borderInfo.getEffect(), particleLocationSW.sub(0, 0, i));
+                    }
+                    for (int i = 0; i <= xLength; i++) {
+                        player.spawnParticles(borderInfo.getEffect(), particleLocationNE.sub(i, 0, 0));
+                        player.spawnParticles(borderInfo.getEffect(), particleLocationSW.add(i + 1, 0, 1));
+                    }
+                });
+            });
+        })).intervalTicks(10).submit(AtherysTowns.getInstance());
+    }
+
+    public void setPlayerViewBorderStatus(Player player, boolean state) throws TownsCommandException {
+        Text.Builder statusText = Text.builder();
+        Text status = state ? Text.of(TextColors.GREEN, "Enabled") : Text.of(TextColors.RED, "Disabled");
+        statusText.append(Text.of("Plot border viewing has been ", status));
+
+        if (state == isPlayerViewingBorders(player)) {
+            throw new TownsCommandException("Plot border viewing is already ", status);
+        }
+        townMsg.info(player, statusText.build());
+        borderViewers.put(player.getUniqueId(), state);
+        if (state) {
+            showAllBorders(player, player.getLocation());
+        } else {
+            clearBorders(player);
+        }
+    }
+
+    public void onPlayerMove(Transform<World> from, Transform<World> to, Player player) {
+        Optional<Plot> plotTo = plotService.getPlotByLocation(to.getLocation());
+        Optional<Plot> plotFrom = plotService.getPlotByLocation(from.getLocation());
+        PlotSelection selection = plotSelectionFacade.getCurrentPlotSelection(player);
+
+        if (isPlayerViewingBorders(player)) {
```

Comment:
This check is pretty cheap to do. We can return early if the player isn't viewing a border. Fetching the `plotTo` `plotFrom` and `selection`  can be moved down